plugins/GSdx/GSRendererDX9.cpp

```diff
@@ -56,6 +56,31 @@ bool GSRendererDX9::CreateDevice(GSDevice* dev)
 	return true;
 }
 
+void GSRendererDX9::EmulateTextureShuffleAndFbmask()
+{
+	size_t count = m_vertex.next;
+	GSVertex* v = &m_vertex.buff[0];
+
+	// Shadow_of_memories_Shadow_Flickering (Okami mustn't call this code)
+	if (m_texture_shuffle && count < 3 && PRIM->FST && (m_context->FRAME.FBMSK == 0)) {
+		// Avious dubious call to m_texture_shuffle on 16 bits games
+		// The pattern is severals column of 8 pixels. A single sprite
+		// smell fishy but a big sprite is wrong.
+
+		// Tomb Raider Angel of Darkness relies on this behavior to produce a fog effect.
+		// In this case, the address of the framebuffer and texture are the same. 
+		// The game will take RG => BA and then the BA => RG of next pixels. 
+		// However, only RG => BA needs to be emulated because RG isn't used.
+		m_texture_shuffle = ((v[1].U - v[0].U) < 256) || m_context->FRAME.Block() == m_context->TEX0.TBP0;
+	}
+
+	if (m_texture_shuffle) {
+		// We can do a partial port for D3D9 that skips the draw call to give it a slight improvement.
+		// It's still broken but more bearable. Broken effect is on the screen but fully instead of vertical lines.
+		throw GSDXRecoverableError();
+	}
```

Comment:
Somehow, it feels like it miss an else part.

Reply:
`else {
		//m_ps_sel.fmt = GSLocalMemory::m_psm[m_context->FRAME.PSM].fmt;

		om_bsel.wrgba = ~GSVector4i::load((int)m_context->FRAME.FBMSK).eq8(GSVector4i::xffffffff()).mask();
	}` Tho that;s part of dx11 only

Reply:
why it is dx11 only ?

Reply:
Oh nvm, I thought it was part of the shader that's for texture shuffle.